src/web/components/ClickToView.test.tsx

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import { render } from '@testing-library/react';
+
+import { ClickToView } from './ClickToView';
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+describe('ClickToView', () => {
+	it('It should render the third party content if it is not tracking', () => {
+		const thirdPartyContent = <div data-testid="third-party-content" />;
+		const { getByTestId } = render(
+			<ClickToView
+				isTracking={false}
+				source="A Third Party"
+				sourceDomain="athirdparty.com"
+			>
+				{thirdPartyContent}
+			</ClickToView>,
+		);
+
+		expect(getByTestId('third-party-content')).toBeInTheDocument();
+	});
+
+	it('It should render a provider specific overlay if a source is present', () => {
+		const { getByText } = render(
+			<ClickToView
+				isTracking={true}
+				source="A Third Party"
+				sourceDomain="athirdparty.com"
+			>
+				<div id="third-party-content" />
+			</ClickToView>,
+		);
+
+		expect(getByText('Allow A Third Party content?')).toBeInTheDocument();
+		expect(
+			getByText(
+				'This article includes content provided by A Third Party. We ask for your permission before anything is loaded, as they may be using cookies and other technologies.',
```

Comment:
If we reduce the length of this string we're still testing that the element shows but perhaps remove some maintainance work for the future?

Reply:
👍